tests/test_nnjai.py

```diff
@@ -0,0 +1,166 @@
+"""
+Tests for the nnjai_wrapp module in the graph_weather package.
+
+This file contains unit tests for AMSUDataset and collate_fn functions.
+"""
+
+from datetime import datetime
+from unittest.mock import MagicMock, patch
+
+import pytest
+import torch
+
+from graph_weather.data.nnja_ai import NNJADataset, collate_fn
+
+@pytest.fixture
+def mock_datacatalog():
+    """
+    Fixture to mock the DataCatalog for unit tests to avoid actual data loading.
+    This mock provides a mock dataset with predefined columns and values.
+    """
+    with patch("graph_weather.data.nnjai_wrapp.DataCatalog") as mock:
+        # Mock dataset structure
+        mock_df = MagicMock()
+        mock_df.columns = ["OBS_TIMESTAMP", "LAT", "LON", "TMBR_00001", "TMBR_00002"]
+
+        # Define a mock row
+        class MockRow:
+            def __getitem__(self, key):
+                data = {
+                    "OBS_TIMESTAMP": datetime.now(),
+                    "LAT": 45.0,
+                    "LON": -120.0,
+                    "TMBR_00001": 250.0,
+                    "TMBR_00002": 260.0,
+                }
+                return data.get(key, None)
+
+        # Configure mock dataset
+        mock_row = MockRow()
+        mock_df.iloc = MagicMock()
+        mock_df.iloc.__getitem__.return_value = mock_row
+        mock_df.__len__.return_value = 100
+
+        mock_dataset = MagicMock()
+        mock_dataset.load_dataset.return_value = mock_df
+        mock_dataset.sel.return_value = mock_dataset
+        mock_dataset.load_manifest = MagicMock()
+
+        mock.return_value.__getitem__.return_value = mock_dataset
+        yield mock
+
+
+def test_sensor_dataset(mock_datacatalog):
+    """
+    Test the SensorDataset class to ensure proper data loading and tensor structure for different sensors.
+    """
+    # Test for AMSU dataset
+    dataset_name = "amsu-1bamua-NC021023"
+    time = datetime(2021, 1, 1, 0, 0)  # Using datetime object instead of string
+    primary_descriptors = ["OBS_TIMESTAMP", "LAT", "LON"]
+    additional_variables = ["TMBR_00001", "TMBR_00002"]
+    dataset = SensorDataset(dataset_name, time, primary_descriptors, additional_variables, sensor_type="AMSU")
```

Comment:
Do these tests still pass? I think they need to be updated for the new names. Ideally, there should be tests for all the different sensor types as well in `NNJADataset`. You should be able to do that with `pytest.parameterize` I think.